tests/test_controller.py

```diff
@@ -25,3 +25,66 @@ def test_init_stac(mocap_nwb, stac_config, rodent_config):
     assert stac.model_cfg == model_cfg
     assert stac._kp_names == sorted_kp_names
     assert isinstance(stac._mj_model, _structs.MjModel)
+
+
+def test_align_joint_dims():
+    from jax import numpy as jp
+
+    types = [0, 1, 2, 3]
+    ranges = [[0.0, 0.0], [0.0, 1.0], [-1.0, 1.0], [-1.0, 1.0]]
+    names = ["root", "balljoint", "slidejoint", "hingejoint"]
+    lb, ub, part_names = _align_joint_dims(types, ranges, names)
+    print(lb)
+
+    true_lb = jp.array(
+        [
+            -jp.inf,
+            -jp.inf,
+            -jp.inf,
+            -jp.inf,
+            -jp.inf,
+            -jp.inf,
+            -jp.inf,
```

Comment:
The bounds of the component values on a unit quaternion are -1, 1, not -inf, +inf. Seems relevant here.